lib/std/math/log_int.zig

```diff
@@ -0,0 +1,79 @@
+const std = @import("../std.zig");
+const math = std.math;
+const testing = std.testing;
+const assert = std.debug.assert;
+const Log2Int = math.Log2Int;
+
+/// Returns the logarithm of `x` for the provided `base`, rounding down to the nearest integer.
+/// Asserts that `base > 1` and `x != 0`.
+pub fn log_int(comptime T: type, base: T, x: T) Log2Int(T) {
+    if (@typeInfo(T) != .Int or @typeInfo(T).Int.signedness != .unsigned)
+        @compileError("log_int requires an unsigned integer, found " ++ @typeName(T));
+
+    assert(base > 1 and x != 0);
```

Comment:
Out of curiousity: Did you test, if the assertion fires, if either base or x is known at comptime and satisfies the subexpressions?

Reply:
I'm not sure I understand the question. Do you mean whether an expression such as `log_int(u8, 3, 82)` panics? In this case, it does not: it returns `@as(u3, 4)`.